Content.Client/Store/Ui/StoreMenu.xaml.cs

```diff
@@ -143,50 +157,98 @@ private void AddListingGui(ListingData listing)
             }
         }
 
-        var newListing = new StoreListingControl(listing, GetListingPriceString(listing), hasBalance, texture);
+        var (listingInStock, discount) = GetListingPriceString(listing, discountData);
+
+        var newListing = new StoreListingControl(listing, listingInStock, discount, hasBalance, texture);
         newListing.StoreItemBuyButton.OnButtonDown += args
             => OnListingButtonPressed?.Invoke(args, listing);
 
         StoreListingsContainer.AddChild(newListing);
     }
 
-    public bool HasListingPrice(Dictionary<ProtoId<CurrencyPrototype>, FixedPoint2> currency, Dictionary<ProtoId<CurrencyPrototype>, FixedPoint2> price)
+    public bool CanBuyListing(Dictionary<ProtoId<CurrencyPrototype>, FixedPoint2> currentBalance, Dictionary<ProtoId<CurrencyPrototype>, FixedPoint2> price, StoreDiscountData? discountData)
     {
-        foreach (var type in price)
+        foreach (var (currency, value) in price)
         {
-            if (!currency.ContainsKey(type.Key))
+            if (!currentBalance.ContainsKey(currency))
                 return false;
 
-            if (currency[type.Key] < type.Value)
+            var amount = value;
+            if (discountData != null && discountData.DiscountAmountByCurrency.TryGetValue(currency, out var discount))
```

Comment:
This ui is being overly complicated for what it's doing.

Reply:
wdym? what is 'excessive' here? U want to move some of that stuff to BUI? Or to system?